boards/seeed/xiao_rp2040/board.cmake

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# This configuration allows selecting what debug adapter debugging to use
+# for the Xiao RP2040 by a command-line argument.
+# It is mainly intended to support both the 'picoprobe' and 'raspberrypi-swd'
+# adapter described in "Getting started with Raspberry Pi Pico".
+# And any other SWD debug adapter might also be usable with this configuration.
+
+# Note that the Xiao RP2040 SWD pins are test points on the underside
+# of the board.
+
+# Set RPI_PICO_DEBUG_ADAPTER to select debug adapter by command-line arguments.
+# e.g.) west build -b rpi_pico -- -DRPI_PICO_DEBUG_ADAPTER=raspberrypi-swd
+# The value is treated as a part of an interface file name that
+# the debugger's configuration file.
+# The value must be the 'stem' part of the name of one of the files
+# in the openocd interface configuration file.
+# The setting is store to CMakeCache.txt.
+if ("${RPI_PICO_DEBUG_ADAPTER}" STREQUAL "")
+	set(RPI_PICO_DEBUG_ADAPTER "cmsis-dap")
+endif()
+
+board_runner_args(openocd --cmd-pre-init "source [find interface/${RPI_PICO_DEBUG_ADAPTER}.cfg]")
```

Comment:
This code looks as a copy of other rp2040 based boards, but with no documentation of how / when to use this variable.

This means that the functionality is a hidden feature.
Please ensure that it's properly documented, or reference to other boards that this exists.
https://docs.zephyrproject.org/latest/boards/raspberrypi/rpi_pico/doc/index.html#using-openocd

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/sparkfun/pro_micro_rp2040/board.cmake

There are other boards based off the pico doing the same thing. I had thought I was supposed to copy the base board in these cases in a similar way? I was basing my own board files in part on how the other RP2040 ones were constructed.

Given this was the normal RP2040 cmake structure I was mostly just leaving it untouched.

Reply:
Ah I think I understand your intent now, apologies, misread the comment initially. You're saying I should add to the index.rst file not a comment here to further document this. Sounds good, will add the original pico related documentation to the Xiao RP2040 one. I don't have one of those jlinks or any more expensive hardware to confirm it works but if it's the same as the other RP2040s I imagine it should (apologies, didn't want to declare it worked in the index.rst without first testing it and since I don't have one).

Reply:
I've included all the additional documentation from the pico for the flashing section here. I can't verify it works for the board as I don't have one of those devices but it should. Let me know if there was another concern here.

Reply:
I've gone ahead and removed the other forms of flashing I can't test myself that I pulled in from those other RP2040 boards. Will defer to others augmenting this in the future who have the tooling to further test it.